### PR TITLE
deeplab AR detect RTD documentation

### LINKED ISSUE #349: missing RTD documentation for deeplab ar detector
add documentation for the command line application in the RTD site
use the same format/layout/style as other apps

### PR DESCRIPTION
**Pull request by elbashandy**
We can use this PR for documenting TECA-BARD as both (TECA-BARD & DeepLabV3+) are AR detection algorithms and they share common arguments.

resolves #349

### CODE REVIEW

**Comment by elbashandy** [comment]
Sure!

**Comment by burlen** [comment]
@taobrienlbl do you have an example batch script from Ankur's deeplab ar detector runs on Cori?

**Comment by taobrienlbl** [comment]
I don't, but let me check with Ankur when I talk to him later this week

**Comment by taobrienlbl** [comment]
Example run on Cori KNL
```
    srun -n 680 -c 16 -N 40 --cpu_bind=cores teca_bayesian_ar_detect \
        --input_regex "/global/project/projectdirs/m1517/cascade/external_datasets/ARTMIP/MERRA_2D/2017/ARTMIP_MERRA_2D_.*\.nc" \
        --cf_reader::t_axis_variable "" \
        --cf_reader::filename_time_template  "ARTMIP_MERRA_2D_%Y%m%d_%H.nc" \
        --steps_per_file 3000 \
        --cf_writer::date_format "%Y" \
        --output_file MERRA2.ar_tag.teca_bard_v1.0.3hourly.%t%.nc4
```

**Review by burlen — APPROVED** [review]
The example for running on cori was not added. However, Ankur's recent talk about the issues in applying the ML detector on datasets it was not trained with I think that there is likely more algorithmic work needed before this code is production ready. I think this documentation is good enough for an experimental app. I think we can merge this now, and I will follow up with some edits that mark it as experimental.

**Comment by burlen** [comment]
I'm making a pass over all app docs. Cherry-picked this into that branch.

File: doc/rtd/applications.rst [hunk 1]
```diff
@@ -666,3 +666,78 @@ regular expression.
         --output_file ${out_dir}/${out_files}                   \
         --arrays PSL TMQ UBOT VBOT U850 V850
 
+Atmospheric Rivers Detection (ARs)
+----------------------------------
+TECA developed detection techniques to quantify the uncertainty of Atmospheric
+River events.
+
+DeepLab v3 Plus
+---------------
```
Comment (burlen) [inline doc/rtd/applications.rst:675]: Since this should be a subsection, - are the wrong markup here. should be ~

File: doc/rtd/applications.rst [hunk 2]
```diff
@@ -666,3 +666,78 @@ regular expression.
...
+    the range of time steps processed.
+
+Example
+~~~~~~~
+.. code-block:: bash
+
+    teca_deeplabv3p_ar_detect                                        \
+        --input_regex "${data_dir}/ARTMIP_MERRA_2D_2017-05.*\.nc$"   \
+        --pytorch_deeplab_model ${model_dir}/cascade_deeplab_IVT.pt  \
+        --output_file test_deeplabv3p_ar_detect_app_output_%t%.nc
```
Comment (burlen) [inline doc/rtd/applications.rst:743]: would be better to have a parallel example showing usage on cori

File: doc/rtd/applications.rst [hunk 3]
```diff
@@ -666,3 +666,78 @@ regular expression.
...
         --arrays PSL TMQ UBOT VBOT U850 V850
 
+Atmospheric Rivers Detection (ARs)
+----------------------------------
+TECA developed detection techniques to quantify the uncertainty of Atmospheric
+River events.
+
+DeepLab v3 Plus
+---------------
+The Deeplabv3+ AR detection application is an MPI+threads parallel code that
```
Comment (burlen) [inline doc/rtd/applications.rst:676]: deeplab isn't MPI+threads. this is a copy paste from the Bayesain AR detcetor.

File: doc/rtd/applications.rst [hunk 4]
```diff
@@ -666,3 +666,78 @@ regular expression.
         --output_file ${out_dir}/${out_files}                   \
         --arrays PSL TMQ UBOT VBOT U850 V850
 
+Atmospheric Rivers Detection (ARs)
+----------------------------------
+TECA developed detection techniques to quantify the uncertainty of Atmospheric
+River events.
```
Comment (burlen) [inline doc/rtd/applications.rst:672]: This introduction doesn't introduce the material that follows very well. this should be revised.

**Review by burlen — COMMENTED** [review]
I've made some comments reflecting some changes that need to be made before this goes live.

**Comment by burlen** [comment]
Moved these commits onto #510

**Pull request closed without merging**